python/mxnet/context.py

```diff
@@ -212,6 +216,14 @@ def gpu(device_id=0):
     return Context('gpu', device_id)
 
 
+def num_gpus():
+    count = ctypes.c_int()
+    check_call(_LIB.MXGetGPUCount(ctypes.byref(count)))
+    return count.value
+
+def gpus():
+    return [gpu(idx) for idx in range(0, num_gpus())]
```

Comment:
can there ever be a case where the gpu device indices are not consecutive or not starting from 0?

Reply:
according to http://docs.nvidia.com/cuda/cuda-runtime-api/group__CUDART__DEVICE.html#group__CUDART__DEVICE_1g159587909ffa0791bbe4b40187a4c6bb this can not be the case: `Sets device as the current device for the calling host thread. Valid device id's are 0 to (cudaGetDeviceCount() - 1).`